src/updatedownloader.cpp

```diff
@@ -177,6 +179,14 @@ void UpdateDownloader::Run()
       UpdateDownloadSink sink(*this, tmpdir);
       DownloadFile(m_appcast.DownloadURL, &sink, this);
       sink.Close();
+
+      if (!SignatureVerifier::DSASHA1SignatureValid(sink.GetFilePath(), m_appcast.DsaSignature))
+      {
+          // remove potentially corrupted file
+          _wremove(sink.GetFilePath().c_str());
```

Comment:
Isn't it better to just call `CleanLeftovers()` (which does full cleanup) here?

Reply:
Sure - haven't seen it.